packages/css/src/plugins/expand-shorthands/text-decoration.tsx

```diff
@@ -0,0 +1,55 @@
+import { Node } from 'postcss-values-parser';
+import { ConversionFunction } from './types';
+import { globalValues, isColor } from './utils';
+
+/**
+ * https://developer.mozilla.org/en-US/docs/Web/CSS/text-decoration
+ */
+export const textDecoration: ConversionFunction = (value) => {
+  const [left, middle, right] = value.nodes;
+  const lineValues = [...globalValues, 'none', 'underline', 'overline', 'line-through', 'blink'];
+  const styleValues = [...globalValues, 'solid', 'double', 'dotted', 'dashed', 'wavy'];
+  const lineValue: string[] = [];
+  let colorValue = '';
+  let styleValue = '';
+
+  /**
+   * Extracts values from a node and mutates variables in scope.
+   * If it returns `true` we should bail out and return no nodes.
+   *
+   * @param node
+   */
+  const extractValues = (node: Node): boolean => {
```

Comment:
nit: Ifs and switches are getting quite complex in `flex-flow.tsx`, `flex.tsx`, `outline.tsx`, `text-decoration.tsx` etc files. I also got confused. Can we simplified it a bit, if possible?

Reply:
i'll see what i can do

Reply:
this is kind of hard to be honest - mainly because of:

1. the order of values can be anything
2. needing to handle invalid usage (doubling up values, missing values, invalid values..)

any suggestions for improvement? would love to make the code easier to reason about

Reply:
Check below [comment](https://github.com/atlassian-labs/compiled/pull/302/files#r506591806) @madou